chart/templates/_helpers.tpl

```diff
@@ -40,6 +40,9 @@ helm.sh/chart: {{ include "joy-generator.chart" . }}
 app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
 {{- end }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- if .Values.podLabels }}
+{{ toYaml .Values.podLabels }}
+{{- end -}}
```

Comment:
Check your expected file, the labels here are not added to the pod, only the meta of resources. In other words the labels were added to the deployment, but not the pod template. 

This block should go below this line https://github.com/nestoca/joy-generator/blob/fa341784e0d0b5b456a130a700f1d127b44b1f10/chart/templates/deployment.yaml#L21